components/status-message/spec/StatusMessage.ts

```diff
@@ -0,0 +1,57 @@
+import { createElement as h } from 'react';
+import { mount } from '@not-govuk/component-test-helpers';
+import StatusMessage from '../src/StatusMessage';
+import {A} from "@not-govuk/components";
+
+describe('StatusMessage', () => {
+  describe('when given a status', () => {
+    const component = mount(h(StatusMessage, { status: 'My heading' }));
+    it('should show the status', () => expect(component.text()).toMatch(/My heading/));
+
+    describe('and extra contents', () => {
+      const component = mount(h(StatusMessage, { status: 'My heading' }, 'extra info'));
+      it('should show the status and the extra contents', () =>
+        expect(component.text()).toMatch(/My heading\s+-\s+extra info/)
+      );
+
+      describe('with an action', () => {
+        const component = mount(h(StatusMessage, {
+          status: 'My heading',
+          actions: h(A, { href: '#' }, 'Run')
+        }, 'extra info'));
+        it('should show the status, extra contents, and action', () =>
+          expect(component.text()).toMatch(/My heading\s+-\s+extra info\s*Run/)
+        );
+      });
+      describe('with multiple actions', () => {
+        const actions = [
+          h(A, { href: '#', key: 1 }, 'Run'),
+          h(A, { href: '#', key: 2 }, 'Jump'),
+          h(A, { href: '#', key: 3 }, 'Sleep')
+        ];
+        const component = mount(h(StatusMessage, { status: 'My heading', actions }, 'extra info'));
+        it('should show the status, extra contents, and actions', () =>
+          expect(component.text()).toMatch(/My heading\s+-\s+extra info\s*Run\s*Jump\s*Sleep/)
+        );
+        it('should show all the actions', () =>
+          expect(component.find('span.hods-status-message__actions a')).toHaveLength(3));
+      });
+    });
+  });
+
+  describe('when given only contents', () => {
+    const component = mount(h(StatusMessage, { }, 'Basic info'));
+    it('should show only the contents', () =>
+      expect(component.text()).toMatch(/Basic info/)
+    );
+  });
+
+  describe('when given only actions', () => {
+    const component = mount(h(StatusMessage, { actions: h(A, { href: '#' }, 'Run') }));
+    it('should not display anything', () => expect(component.text()).toEqual(''));
```

Comment:
Does this scenario make much sense?

Why not just make the `summary` prop mandatory?

Reply:
Even if the prop is mandatory it could still be empty. This is intended to be a basic defence against an empty blue box.

Reply:
I've allowed for the status to be specified by providing either the `status` prop, or children. If both were empty I suppose I could throw an error, but I think it's probably better to just not display anything :shrug: I could take the _"list items don't have unique key prop"_ error example from React rendering, and just print a warning to the console (but otherwise continue to successfully render nothing)?

Reply:
I don't think you should defend against that.

An empty blue box is more intuitive than nothing at all.

I'd make props mandatory where possible but not defend against empty strings. If the user gives you junk that's their fault.

I'm not sure I'd do the thing with `children` either. If you have children, I'd try to use them for one purpose, if possible.

Note that `children` isn't just text it can be more React components.

Reply:
The use of children there _is_ by design. The `status` should just be something short, but the children could have more varied content.

Reply:
In that case, how about we make status mandatory and children will be optional? (Rather than allowing people to place the status in the children.)

Reply:
Agreed. Fixed